### PR TITLE
feat: redis bridge v2

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes https://emqx.atlassian.net/browse/EMQX-11153

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: .ci/docker-compose-file/redis/sentinel-tls/sentinel-base.conf [hunk 1]
```diff
@@ -8,7 +8,7 @@ tls-key-file /etc/certs/key.pem
 tls-ca-cert-file /etc/certs/cacert.pem
 tls-auth-clients no
 
-sentinel monitor mymaster redis-sentinel-tls-master 6389 1
-sentinel auth-pass mymaster public
-sentinel down-after-milliseconds mymaster 10000
-sentinel failover-timeout mymaster 20000
```
Comment (zhongwencool) [inline .ci/docker-compose-file/redis/sentinel-tls/sentinel-base.conf:14]: Split mymaster to mytcpmaster and mytlsmaster for CI confilt.

File: apps/emqx_bridge/src/schema/emqx_bridge_v2_schema.erl [hunk 2]
```diff
@@ -125,6 +127,28 @@ registered_schema_fields() ->
      || {_BridgeV2Type, Module} <- emqx_action_info:registered_schema_modules()
     ].
 
+make_producer_action_schema(ActionParametersRef) ->
```
Comment (zhongwencool) [inline apps/emqx_bridge/src/schema/emqx_bridge_v2_schema.erl:130]: I copy this from master.

File: apps/emqx_bridge_redis/src/emqx_bridge_redis.erl [hunk 3]
```diff
@@ -141,34 +160,37 @@ method_fields(put, ConnectorType) ->
...
+            ?HOCON(
+                ?R_REF(action_parameters),
+                #{
+                    required => true,
+                    desc => ?DESC(producer_action)
+                }
+            )
+        ),
+    [ResourceOpts] = resource_fields(Type),
+    lists:keyreplace(resource_opts, 1, Schema, ResourceOpts).
```
Comment (zhongwencool) [inline apps/emqx_bridge_redis/src/emqx_bridge_redis.erl:187]: redis cluster don't support batch operation, so we replace this.

File: apps/emqx_resource/src/emqx_resource_manager.erl [hunk 4]
```diff
@@ -693,6 +693,9 @@ stop_resource(#data{state = ResState, id = ResId} = Data) ->
     ok = emqx_metrics_worker:reset_metrics(?RES_METRICS, ResId),
     NewData#data{status = stopped}.
 
+is_init_state(#data{state = ResState, status = Status}) ->
+    ResState =:= undefined andalso Status =:= connecting.
```
Comment (zhongwencool) [inline apps/emqx_resource/src/emqx_resource_manager.erl:697]: Clear resource(stop instance) even ResState is undefined.

File: scripts/pre-compile.sh [hunk 5]
```diff
@@ -27,10 +27,20 @@ cd -P -- "$(dirname -- "${BASH_SOURCE[0]}")/.."
...
+  start=$(date +%s)
+  curl -L --silent --show-error \
+       --output "apps/emqx_dashboard/priv/desc.zh.hocon" \
+       "https://raw.githubusercontent.com/emqx/emqx-i18n/${I18N_REPO_BRANCH}/desc.zh.hocon"
+  end=$(date +%s)
+  duration=$(echo "$end $start" | awk '{print $1 - $2}')
+  echo "updated  i18n file using $duration seconds, set UPDATE_I18N=false to skip"
+else
+  echo "skipping update i18n file from emqx-i18n repo, set UPDATE_I18N=true to update"
+fi
```
Comment (zhongwencool) [inline scripts/pre-compile.sh:43]: Downloading the i18n file consumes too much time in daily development, 
especially when the network is poor. So make an option not to download.

**Comment by thalesmg** [comment]
Looking at the epic/ticket, it seems this is meant for e5.4.0, so maybe we need to change the base branch to `master` (or `release-54`) if that information is accurate.  :thinking:

**Comment by zhongwencool** [comment]
> Looking at the epic/ticket, it seems this is meant for e5.4.0, so maybe we need to change the base branch to `master` (or `release-54`) if that information is accurate. 🤔

Got it!

**Pull request closed without merging**